features/251-templates/index.md

```diff
@@ -0,0 +1,177 @@
+  TemplateConfigurationTypeUserVerificationErrorHTML          TemplateConfigurationType = "user_verification_error.html"          // Keep URL option
+)
+```
+
+### Template Language Tag
+
+Each template may optionally have a language tag. In the future, we may support sending localized message. The language tag is specified in [BCP47](https://tools.ietf.org/html/bcp47).
+
+### Template Key
+
+If the developer has configured `email` and `secondary_email` as login IDs. User Verification supports verifying both with different templates. Therefore, `key` is required to differentiate different instances of the same type of the template.
+
+## The `templates` directory next to `skygear.yaml`
+
+The developer organizes their templates in the `templates` directory next to `skygear.yaml`.
+
+The templates must be organized in the following way in order to be recognized.
+
+```
+templates[/<language-tag>][/<key>]/<type>
+```
+
+### Examples
+
+#### Example 1
+
+```
+templates/welcome_email.txt
+templates/welcome_email.html
+
+templates/forgot_password_email.txt
+templates/forgot_password_email.html
+
+templates/forgot_password_reset.html
+templates/forgot_password_success.html
+templates/forgot_password_error.html
+
+templates/user_verification_general_error.html
+
+templates/email/user_verification_message.txt
+templates/email/user_verification_message.html
+templates/email/user_verification_success.html
+templates/email/user_verification_error.html
+
+templates/phone/user_verification_message.txt
+templates/phone/user_verification_success.html
+templates/phone/user_verification_error.html
+```
+
+Given the above hierarchy, the following templates are recognized:
+
+- Plain text and HTML Welcome email
+- Plain text and HTML Forgot Password email
+- Reset, success and error HTML page of resetting password
+- General error HTML Page of User Verification
+- Plain text and HTML verification email of login ID `email`
+- SMS verification message of login ID `phone`
+
+#### Example 2
+
+```
+templates/welcome_email.txt
+templates/welcome_email.html
+
+templates/zh-Hant/welcome_email.txt
+templates/zh-Hant/welcome_email.html
+
+templates/zh-Hans/welcome_email.txt
+templates/zh-Hans/welcome_email.html
+
+templates/ja/welcome_email.txt
+templates/ja/welcome_email.html
+```
+
+Given the above hierarchy, the developer provides:
+
+- Plain text and HTML Welcome email without language tag, which is the fallback
+- Plain text and HTML Welcome email in Traditional Chinese
+- Plain text and HTML Welcome email in Simplified Chinese
+- Plain text and HTML Welcome email in Japanese
+
+## Language Tag Matching
+
+The matching is done by the package `"golang.org/x/text/language"`.
+
+## Integration with skycli
+
+### `skycli config list-templates`
+
+It lists the remote templates. That is, the templates that are previously uploaded.
+
+#### Sample Output
+
+```
+templates/welcome_email.txt
+templates/welcome_email.html
+```
+
+### `skycli config update-templates [-y]`
```

Comment:
I think to keep it simple, we will not have a command to update single template, right? If so, I think we need the `download-templates` command that you mentioned.

For some implementation detail, so portal and cli may use different controller api to update templates?

Reply:
For portal, we need an API to PUT multiple templates since the current UI allows the developer to edit multiple templates with a tabbed editor.

For cli, we need an API to replace the whole set of templates.